extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1367,6 +1482,54 @@ export class Project implements ISqlProject {
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Sets the value of project property, overrides value if already present in project file.
+	 * @param propertyName Name of property
+	 * @param propertyValue Value of property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string) {
+		const propertyElements = this.projFileXmlDoc!.documentElement.getElementsByTagName(propertyName);
+
+		let propertyElement: Element | undefined;
+		if (propertyElements.length === 0) {
+			propertyElement = this.addProjectPropertyTag(propertyName);
+		} else {
+			propertyElement = propertyElements[0];
+			if (propertyElement.childNodes.length > 0) {
+				propertyElement.removeChild(propertyElement.childNodes[0]);
+			}
+		}
+
+		propertyElement?.appendChild(this.projFileXmlDoc!.createTextNode(propertyValue));
+		await this.serializeToProjFile(this.projFileXmlDoc);
+	}
+
+	/**
+	 * Adds empty project property tag
+	 * @param propertyTag Tag to add
+	 * @returns Added HTMLElement tag
+	 */
+	private addProjectPropertyTag(propertyTag: string): HTMLElement | undefined {
+		let propertyGroup = this.projFileXmlDoc!.getElementsByTagName(constants.PropertyGroup).item(0);
```

Comment:
Why do we have to always do `!` for the `projFileXmlDoc`? Can't we just check if it's undefined or not at the beginning of the method and return, if there is nothing for us to store the property in?